segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -272,17 +277,21 @@ CompletableFuture<Void> process(Operation operation, OperationPriority priority)
                 result.completeExceptionally(e);
             }
 
-            if (priority.isThrottlingExempt()) {
-                // A throttle-exempt operation has just been added (these are time-critical operations, which must execute
-                // right away). If there is a throttling delay in progress, we must abort it, otherwise this operation
-                // may not get a chance to execute.
-                getThrottler().notifyThrottleSourceChanged();
-            }
+            // A throttle-exempt operation has just been added (these are time-critical operations, which must execute
+            // right away). If there is a throttling delay in progress, we must abort it, otherwise this operation
+            // may not get a chance to execute.
+            notifyThrottleSourceChanged(priority::isThrottlingExempt);
         }
 
         return result;
     }
 
+    private void notifyThrottleSourceChanged(Supplier<Boolean> condition) {
```

Comment:
This seems an excessive level of abstraction. There are only two places where this is called and the method is just an if statement followed by another invocation, but because the callers need a different conditional they pass in their condition as a function. (Which is odd because they could just pass in a boolean) 

It would be more readable if this method were simply inlined.

Reply:
I have removed that method and called the method inside `if` blocks.